frontend/src/app/main/data/workspace/transforms.cljs

```diff
@@ -913,13 +918,13 @@
                   (pcb/update-shapes moving-shapes-ids ctl/remove-layout-item-data))
                 ;; Remove component-root property when moving a shape inside a component
                 (cond-> (ctn/get-instance-root objects frame)
-                  (pcb/update-shapes moving-shapes-ids #(dissoc % :component-root)))
+                  (pcb/update-shapes moving-shapes-children-ids #(dissoc % :component-root)))
                 ;; Add component-root property when moving a component outside a component
                 (cond-> (not (ctn/get-instance-root objects frame))
-                  (pcb/update-shapes moving-shapes-ids (fn [shape]
-                                                         (if (ctk/instance-head? shape)
-                                                           (assoc shape :component-root true)
-                                                           shape))))
+                  (pcb/update-shapes moving-shapes-children-ids (fn [shape]
+                                                                  (if (ctk/instance-head? shape)
+                                                                    (assoc shape :component-root true)
+                                                                    shape))))
```

Comment:
Confirmed. If you have a component copy with some other copies nested, and move it out of the main, it crashes because it tries to add component-root to all heads.

Reply:
Ops, you are right! thanks!